tests/Unit/CssInlinerTest.php

```diff
@@ -2982,6 +2982,64 @@ public function inlineCssNotInDebugModeKeepsInvalidOrUnrecognizedSelectorsInMedi
         self::assertContainsCss($css, $subject->render());
     }
 
+    /**
+     * @test
+     */
+    public function htmlWithoutHeadOrBodyElementsAndStylesCopiedToStyleBlocksIsProperlyInlined()
+    {
+        $testHTML = '    
+<div id="newsletter-display-problems">
+    <a href="">
+        Falls der Newsletter nicht korrekt angezeigt wird, klicken Sie bitte hier.
+    </a>
+</div>
+<div id="newsletter-main-content-container">
+<!--TYPO3SEARCH_begin-->
+<!--DMAILER_SECTION_BOUNDARY_-->
+<div id="c7001" class="frame frame-default frame-type-textmedia frame-layout-0">
+    <header>
+        <h1 class="">
+			Newsletter Beispiel
+        </h1>
+    </header>
+<div class="ce-textpic ce-center ce-above">
+    <div class="ce-bodytext">
+        <h2>Lorem ipsum</h2>
+        <p>
+            Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
+        </p>
+        <hr />
+        <h2>Lorem ipsum</h2>
+        <p>
+            Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.&nbsp;Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
+        </p>
+        <p>
+            <a href="/fileadmin/_processed_/c/a/test.jpg" class="lightbox" rel="lightbox[922]" data-fancybox="images"><img height="198" src="fileadmin/_processed_/c/a/test.jpg" width="300" /></a>
+            &nbsp;
+            <a href="/fileadmin/_processed_/c/2/test.jpg" class="lightbox" rel="lightbox[922]" data-fancybox="images"><img height="198" src="fileadmin/_processed_/c/2/test.jpg" width="300" /></a>
+            &nbsp;
+            <a href="/fileadmin/_processed_/3/0/test.jpg" class="lightbox" rel="lightbox[922]" data-fancybox="images"><img height="198" src="fileadmin/_processed_/3/0/test.jpg" width="300" /></a>
+        </p>
+    </div>
+</div>
```

Comment:
Could you please reduce the HTML to the bare minimum needed for this test to be meaningful? This will help readers see what's relevant (and also make the test shorter).

Reply:
I did reduce both the css and the html to a bare minimum now